pkg/invariants/incluster_disruption_serializer/remote.go

```diff
@@ -381,18 +255,177 @@ func fetchEventsFromFileOnNode(ctx context.Context, clientset *kubernetes.Client
+	if err != nil {
+		return err
+	}
+	configClient, err := configclient.NewForConfig(i.adminRESTConfig)
+	if err != nil {
+		return err
+	}
+	infra, err := configClient.ConfigV1().Infrastructures().Get(context.Background(), "cluster", metav1.GetOptions{})
+	if err != nil {
+		return err
+	}
+
+	internalAPI, err := url.Parse(infra.Status.APIServerInternalURL)
+	if err != nil {
+		return err
+	}
+	apiIntHost := internalAPI.Hostname()
+
+	err = createNamespace(ctx, kubeClient)
+	if err != nil {
+		return err
+	}
+	err = createServiceAccount(ctx, kubeClient)
+	if err != nil {
+		return err
+	}
+	err = createRBACPrivileged(ctx, kubeClient)
+	if err != nil {
+		return err
+	}
+	err = createMonitorRole(ctx, kubeClient)
+	if err != nil {
+		return err
+	}
+	err = createMonitorCRB(ctx, kubeClient)
+	if err != nil {
+		return err
+	}
+	err = createServiceNetworkDS(ctx, kubeClient)
+	if err != nil {
+		return err
+	}
+	err = createLocalhostDS(ctx, kubeClient)
+	if err != nil {
+		return err
+	}
+	err = createInternalLBDS(ctx, kubeClient, apiIntHost)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func (i *InvariantInClusterDisruption) CollectData(ctx context.Context, storageDir string, beginning time.Time, end time.Time) (monitorapi.Intervals, []*junitapi.JUnitTestCase, error) {
+	fmt.Fprintf(os.Stderr, "Stopping in-cluster monitoring daemonsets\n")
+	client, err := kubernetes.NewForConfig(i.adminRESTConfig)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	// Start watcher before daemonset removal request to make sure delete event is not skipped
+	c := make(chan error)
+	timeLimitedCtx, cancel := context.WithTimeout(ctx, 5*time.Minute)
+	defer cancel()
+	go func() {
+		_, watchErr := watchtools.UntilWithSync(timeLimitedCtx,
+			cache.NewListWatchFromClient(
+				client.AppsV1().RESTClient(), "daemonsets", namespace, fields.Everything()),
+			&appsv1.DaemonSet{},
+			nil,
+			func(event watch.Event) (bool, error) {
+				return event.Type == watch.Deleted, nil
+			},
+		)
+		if watchErr != nil {
+			watchErr = fmt.Errorf("daemonsets in namespace %s didn't get destroyed: %v", namespace, watchErr)
+		}
+		c <- watchErr
+	}()
+
+	// Remove daemonsets first to avoid trailing false-positive disruption intervals
+	dsClient := client.AppsV1().DaemonSets(namespace)
+	err = dsClient.DeleteCollection(ctx, metav1.DeleteOptions{}, metav1.ListOptions{})
+	if err != nil && !apierrors.IsNotFound(err) {
+		return nil, nil, fmt.Errorf("error removing daemonsets in namespace %s: %v", namespace, err)
+	}
+	if watchErr := <-c; watchErr != nil {
+		return nil, nil, watchErr
+	}
+
+	fmt.Fprintf(os.Stderr, "Fetching collected in-cluster disruption data\n")
+	nodes, err := client.CoreV1().Nodes().List(ctx, metav1.ListOptions{})
+	if err != nil {
+		return nil, nil, err
+	}
+	var events monitorapi.Intervals
+	var errs []error
+	for _, node := range nodes.Items {
+		nodeEvents, err := fetchNodeInClusterEvents(ctx, client, &node)
```

Comment:
actually, thinking about this overall, why not use a localvolume mount to store disruption log lines, then simply cat the existing file on startup and do a get pod log instead of trying to access via the kubelet?

Reply:
iiuc we can move collectdata to run as a pod, but that would still require the same calls to kubelet - and additional call to get pod logs